pkg/scaling/scaledjob/scale_metrics.go

```diff
@@ -0,0 +1,181 @@
+	var queueLength int64
+	var maxValue int64
+	isActive := false
+
+	logger := logf.Log.WithName("scalemetrics")
+	scalersMetrics := getScalersMetrics(ctx, scalers, scaledJob, logger, recorder)
+	switch scaledJob.Spec.ScalingStrategy.MultipleScalersCalculation {
+	case "min":
+		for _, metrics := range scalersMetrics {
+			if (queueLength == 0 || metrics.queueLength < queueLength) && metrics.isActive {
+				queueLength = metrics.queueLength
+				maxValue = metrics.maxValue
+				isActive = metrics.isActive
+			}
+		}
+	case "avg":
+		queueLengthSum := int64(0)
+		maxValueSum := int64(0)
+		length := 0
+		for _, metrics := range scalersMetrics {
+			if metrics.isActive {
+				queueLengthSum += metrics.queueLength
+				maxValueSum += metrics.maxValue
+				isActive = metrics.isActive
+				length++
+			}
+		}
+		if length != 0 {
+			queueLength = divideWithCeil(queueLengthSum, int64(length))
+			maxValue = divideWithCeil(maxValueSum, int64(length))
+		}
+	case "sum":
+		for _, metrics := range scalersMetrics {
+			if metrics.isActive {
+				queueLength += metrics.queueLength
+				maxValue += metrics.maxValue
+				isActive = metrics.isActive
+			}
+		}
+	default: // max
+		for _, metrics := range scalersMetrics {
+			if metrics.queueLength > queueLength && metrics.isActive {
+				queueLength = metrics.queueLength
+				maxValue = metrics.maxValue
+				isActive = metrics.isActive
+			}
+		}
+	}
+	maxValue = min(scaledJob.MaxReplicaCount(), maxValue)
+	logger.V(1).WithValues("ScaledJob", scaledJob.Name).Info("Checking if ScaleJob scalers are active", "isActive", isActive, "maxValue", maxValue, "MultipleScalersCalculation", scaledJob.Spec.ScalingStrategy.MultipleScalersCalculation)
+
+	return isActive, queueLength, maxValue
+}
+
+func getScalersMetrics(ctx context.Context, scalers []scalers.Scaler, scaledJob *kedav1alpha1.ScaledJob, logger logr.Logger, recorder record.EventRecorder) []scalerMetrics {
+	scalersMetrics := []scalerMetrics{}
+
+	// TODO refactor this, do chores, reduce the verbosity ie: V(1) and frequency of logs
+	// move relevant funcs getTargetAverageValue(), min() and divideWithCeil() out of scaler_handler.go
+	for _, scaler := range scalers {
+		var queueLength int64
+		var targetAverageValue int64
+		isActive := false
+		maxValue := int64(0)
+		scalerType := fmt.Sprintf("%T:", scaler)
+
+		scalerLogger := logger.WithValues("ScaledJob", scaledJob.Name, "Scaler", scalerType)
+
+		metricSpecs := scaler.GetMetricSpecForScaling()
+
+		// skip scaler that doesn't return any metric specs (usually External scaler with incorrect metadata)
+		// or skip cpu/memory resource scaler
+		if len(metricSpecs) < 1 || metricSpecs[0].External == nil {
+			continue
+		}
+
+		isTriggerActive, err := scaler.IsActive(ctx)
+
+		scalerLogger.Info("Active trigger", "isTriggerActive", isTriggerActive)
+
+		targetAverageValue = getTargetAverageValue(metricSpecs)
+
+		scalerLogger.Info("Scaler targetAverageValue", "targetAverageValue", targetAverageValue)
```

Comment:
These two logging messages can be merged into one, together with the one below:  
https://github.com/kedacore/keda/pull/2016/files#diff-4df7921990b111ce156bcf67689749dfc9c321fd5bdf2d68fa5f94f1a2a92c37R119

Reply:
How about this? https://github.com/kedacore/keda/pull/2016/commits/cddf760f7b424c9c970717738bc447cf43460f47